### PR TITLE
Resolve #761, #1246, #1413, #1426, #1291

### LINKED ISSUE #761: Support more ERC20
**The following coin will be added after discussion**
USDT
https://etherscan.io/token/0xdac17f958d2ee523a2206206994597c13d831ec7

USDC
https://etherscan.io/token/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48

TUSD
https://etherscan.io/token/0x0000000000085d4780B73119b644AE5ecd22b376
___
**The following coin will NOT be added at current state**
BNB
https://etherscan.io/token/0xB8c77482e45F1F44dE1745F52C74426C631bDD52

LEO
https://etherscan.io/token/0x2af5d2ad76741191d15dfe7bf6ac92d4bd912ca3

HT
https://etherscan.io/token/0x6f259637dcd74c767781e37bc6133cd6a68aa161

LINK
https://etherscan.io/token/0x514910771af9ca656af840dff83e8264ecf986ca

MKR
https://etherscan.io/token/0x9f8f72aa9304c8b593d555f12ef6589cc3a579a2

CRO
https://etherscan.io/token/0xa0b73e1ff0b80914ab6fe0444e65848c4c34450b

OKB
https://etherscan.io/token/0x75231f58b43240c9718dd58b4967c5114342a86c

BAT
https://etherscan.io/token/0x0d8775f648430679a709e98d2b0cb6250d2887ef

REP
https://etherscan.io/token/0x1985365e9f78359a9B6AD760e32412f4a445E862

ZRX
https://etherscan.io/token/0xe41d2489571d322189246dafa5ebde1f4699f498

KNC
https://etherscan.io/token/0xdd974d5c2e2928dea5f71b9825b8b646686bd200

WBTC
https://etherscan.io/token/0x2260fac5e5542a773aa44fbcfedf7c193bc2c599

### LINKED ISSUE #1246: Make all address clickable

### PR DESCRIPTION
**Pull request by InVincible2016**

### CODE REVIEW

File: src/tokens.js [hunk 1]
```diff
@@ -2,106 +2,140 @@ import EthereumLogo from './images/eth.png'
...
   {
-    platformType:'bitcoin',
+    platformType: 'bitcoin',
     cryptoType: 'bitcoin',
     title: 'Bitcoin',
     symbol: 'BTC',
     logo: BitcoinLogo,
     decimals: 8,
-    txFeesCryptoType: 'bitcoin'
+    txFeesCryptoType: 'bitcoin',
```
Comment (timothywangdev) [inline src/tokens.js:18]: no comma

File: src/tokens.js [hunk 2]
```diff
@@ -2,106 +2,140 @@ import EthereumLogo from './images/eth.png'
...
   {
-    platformType:'ethereum',
+    platformType: 'ethereum',
     cryptoType: 'ethereum',
     title: 'Ethereum',
     symbol: 'ETH',
     logo: EthereumLogo,
     decimals: 18,
-    txFeesCryptoType: 'ethereum'
+    txFeesCryptoType: 'ethereum',
```
Comment (timothywangdev) [inline src/tokens.js:27]: no comma

File: src/tokens.js [hunk 3]
```diff
@@ -2,106 +2,140 @@ import EthereumLogo from './images/eth.png'
...
+    platformType: 'ethereum',
     cryptoType: 'dai',
     title: 'Dai',
     symbol: 'DAI',
     logo: DaiLogo,
-    address: process.env.REACT_APP_DAI_ADDRESS,
+    address: erc20TokensAddress.dai,
     decimals: 18,
-    txFeesCryptoType: 'ethereum'
+    txFeesCryptoType: 'ethereum',
```
Comment (timothywangdev) [inline src/tokens.js:37]: no comma

File: src/tokens.js [hunk 4]
```diff
@@ -2,106 +2,140 @@ import EthereumLogo from './images/eth.png'
...
-    platformType:'libra',
+    platformType: 'libra',
     cryptoType: 'libra',
     title: 'Libra',
     symbol: 'Libra',
     logo: LibraLogo,
     decimals: 6,
-    txFeesCryptoType: 'libra'
-  }
+    txFeesCryptoType: 'libra',
```
Comment (timothywangdev) [inline src/tokens.js:46]: no comma

File: src/tokens.js [hunk 5]
```diff
@@ -2,106 +2,140 @@ import EthereumLogo from './images/eth.png'
...
+  },
+  {
+    platformType: 'ethereum',
+    cryptoType: 'usd-coin',
+    title: 'USD Coin',
+    symbol: 'USDC',
+    logo: UsdcLogo,
+    address: erc20TokensAddress['usd-coin'],
+    decimals: 6,
+    txFeesCryptoType: 'ethereum',
```
Comment (timothywangdev) [inline src/tokens.js:56]: no comma

File: src/tokens.js [hunk 6]
```diff
@@ -2,106 +2,140 @@ import EthereumLogo from './images/eth.png'
...
+  },
+  {
+    platformType: 'ethereum',
+    cryptoType: 'tether',
+    title: 'Tether',
+    symbol: 'USDT',
+    logo: UsdtLogo,
+    address: erc20TokensAddress['tether'],
+    decimals: 6,
+    txFeesCryptoType: 'ethereum',
```
Comment (timothywangdev) [inline src/tokens.js:66]: no comma

File: src/tokens.js [hunk 7]
```diff
@@ -2,106 +2,140 @@ import EthereumLogo from './images/eth.png'
...
+  },
+  {
+    platformType: 'ethereum',
+    cryptoType: 'true-usd',
+    title: 'TrueUSD',
+    symbol: 'TUSD',
+    logo: TusdLogo,
+    address: erc20TokensAddress['true-usd'],
+    decimals: 18,
+    txFeesCryptoType: 'ethereum',
```
Comment (timothywangdev) [inline src/tokens.js:76]: no comma

File: src/tokens.js [hunk 8]
```diff
@@ -2,106 +2,140 @@ import EthereumLogo from './images/eth.png'
...
+  },
 ]
 
 export function getCryptoTitle (cryptoType) {
-  const c = cryptoSelections.find(crypto => {
+  const c = cryptoSelections.find((crypto) => {
```
Comment (timothywangdev) [inline src/tokens.js:81]: should have no changes for the following code

File: src/wallets/walletConnect.js [hunk 9]
```diff
@@ -78,15 +82,15 @@ export default class WalletConnect implements IWallet<AccountData> {
       receivable: true,
       sendable: true,
 
-      lastSynced: 0
+      lastSynced: 0,
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:85]: no comma

File: src/wallets/walletConnect.js [hunk 10]
```diff
@@ -20,14 +21,17 @@ export default class WalletConnect implements IWallet<AccountData> {
     // Create a walletConnector
     if (!window.walletConnector) {
       window.walletConnector = new _WalletConnect({
-        bridge: 'https://bridge.walletconnect.org' // Required
+        bridge: 'https://bridge.walletconnect.org', // Required
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:24]: no comma

File: src/wallets/metamask.js [hunk 11]
```diff
@@ -152,22 +160,22 @@ export default class MetamaskWallet implements IWallet<AccountData> {
...
   }
 
   getTxFee = async ({
     value,
-    options
+    options,
   }: {
     value: BasicTokenUnit,
-    options?: Object
+    options?: Object,
```
Comment (timothywangdev) [inline src/wallets/metamask.js:172]: no comma

File: src/wallets/metamask.js [hunk 12]
```diff
@@ -152,22 +160,22 @@ export default class MetamaskWallet implements IWallet<AccountData> {
...
   }): Promise<TxFee> => {
     const accountData = this.getAccount().getAccountData()
     return WalletUtils.getTxFee({
       value,
       cryptoType: accountData.cryptoType,
-      directTransfer: !!options && options.directTransfer
+      directTransfer: !!options && options.directTransfer,
```
Comment (timothywangdev) [inline src/wallets/metamask.js:178]: no comma

File: src/wallets/metamask.js [hunk 13]
```diff
@@ -152,22 +160,22 @@ export default class MetamaskWallet implements IWallet<AccountData> {
 
     // boardcast tx
     return {
-      txHash: await WalletUtils.web3SendTransactions(window._web3.eth.sendTransaction, txObj)
+      txHash: await WalletUtils.web3SendTransactions(window._web3.eth.sendTransaction, txObj),
```
Comment (timothywangdev) [inline src/wallets/metamask.js:163]: no comma

File: src/wallets/metamask.js [hunk 14]
```diff
@@ -125,12 +128,12 @@ export default class MetamaskWallet implements IWallet<AccountData> {
...
     value,
     txFee,
-    options
+    options,
   }: {
     to: Address,
     value: BasicTokenUnit,
     txFee: TxFee,
-    options?: Object
+    options?: Object,
```
Comment (timothywangdev) [inline src/wallets/metamask.js:136]: no comma

File: src/wallets/metamask.js [hunk 15]
```diff
@@ -125,12 +128,12 @@ export default class MetamaskWallet implements IWallet<AccountData> {
     to,
     value,
     txFee,
-    options
+    options,
```
Comment (timothywangdev) [inline src/wallets/metamask.js:131]: no comma

File: src/wallets/metamask.js [hunk 16]
```diff
@@ -85,15 +88,15 @@ export default class MetamaskWallet implements IWallet<AccountData> {
       receivable: true,
       sendable: true,
 
-      lastSynced: 0
+      lastSynced: 0,
```
Comment (timothywangdev) [inline src/wallets/metamask.js:91]: no comma

File: src/wallets/walletConnect.js [hunk 17]
```diff
@@ -224,7 +228,7 @@ export default class WalletConnect implements IWallet<AccountData> {
       await window.walletConnector.killSession()
     }
     window.walletConnector = new _WalletConnect({
-      bridge: 'https://bridge.walletconnect.org' // Required
+      bridge: 'https://bridge.walletconnect.org', // Required
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:231]: no comma

File: src/wallets/walletConnect.js [hunk 18]
```diff
@@ -204,7 +208,7 @@ export default class WalletConnect implements IWallet<AccountData> {
     txObj = {
       ...txObj,
       gas: txFee.gas,
-      gasPrice: txFee.price
+      gasPrice: txFee.price,
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:211]: no comma

File: src/wallets/walletConnect.js [hunk 19]
```diff
@@ -182,16 +186,16 @@ export default class WalletConnect implements IWallet<AccountData> {
...
     value: BasicTokenUnit,
-    options?: Object
+    options?: Object,
   }): Promise<TxFee> => {
     const accountData = this.getAccount().getAccountData()
     return WalletUtils.getTxFee({
       value,
       cryptoType: accountData.cryptoType,
-      directTransfer: !!options && options.directTransfer
+      directTransfer: !!options && options.directTransfer,
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:198]: no comma

File: src/wallets/walletConnect.js [hunk 20]
```diff
@@ -182,16 +186,16 @@ export default class WalletConnect implements IWallet<AccountData> {
 
   getTxFee = async ({
     value,
-    options
+    options,
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:189]: no comma

File: src/wallets/walletConnect.js [hunk 21]
```diff
@@ -182,16 +186,16 @@ export default class WalletConnect implements IWallet<AccountData> {
...
   }: {
     value: BasicTokenUnit,
-    options?: Object
+    options?: Object,
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:192]: no comma

File: src/wallets/walletConnect.js [hunk 22]
```diff
@@ -147,12 +151,12 @@ export default class WalletConnect implements IWallet<AccountData> {
...
     value,
     txFee,
-    options
+    options,
   }: {
     to: Address,
     value: BasicTokenUnit,
     txFee: TxFee,
-    options?: Object
+    options?: Object,
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:159]: no comma

File: src/wallets/walletConnect.js [hunk 23]
```diff
@@ -147,12 +151,12 @@ export default class WalletConnect implements IWallet<AccountData> {
     to,
     value,
     txFee,
-    options
+    options,
```
Comment (timothywangdev) [inline src/wallets/walletConnect.js:154]: no comma

File: src/components/MicroComponents/ClickableAddress.jsx [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+    display: 'flex',
+    alignItems: 'center'
+  }
+})
+
+const ClickableAddress = (props: ClickableAddressProps) => {
+  const { children, typographyProps } = props
+  if (!children) return null
+  const classes = styles()
+  const cryptoType = children.startsWith('0x') ? 'ethereum' : 'bitcoin'
```
[the remark this answers is not in the document]
Comment (InVincible2016) [reply src/components/MicroComponents/ClickableAddress.jsx:25]: For now, this is working fine in our cases. This makes this component take in fewer props.

[the remark this answers is not in the document]
Comment (timothywangdev) [reply src/components/MicroComponents/ClickableAddress.jsx:25]: add a todo comment then

[the remark this answers is not in the document]
Comment (timothywangdev) [reply src/components/MicroComponents/ClickableAddress.jsx:25]: // TODO: ...

**Pull request merged**